docs/api.md

```diff
@@ -6,6 +6,206 @@ challenges presented by JavaScript and web browsers.
 Please direct clarifications or observations of inaccuracy to [our issue
 tracker](https://github.com/iodide-project/iodide/issues/new).
 
+
+## `iodide.file`
+
+The `iodide.file` API provides convenience functions around working with files
+uploaded to the Iodide server in your notebook.
+
+
+
+### `iodide.file.save(data, fileName[, saveOptions])`
+
+Returns a
+[Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises),
+which, when resolved, will signal that `data` was uploaded to the server under
+the filename `fileName`.
+
+`data` (required) is any object or variable in the eval name space. It can be
+anything, really - an array of data, or a csv, or a binary blob.
+
+`fileName` (required) is a string that represents the file name.
+
+The optional argument `saveOptions` has the following keys:
+
+- `overwrite` (optional, default `false`): if `true`, will overwrite whatever is
+  at `fileName` with `data`. If `false`, will throw an error and halt the
+  subsequent enqueued evaluation (similar to a syntax error).
+
+
+#### `iodide.file.save` patterns
+
+```javascript
+
+// upload a csv to the server. Assume the object csvData is an array of objects
+// representing a columnar dataset. This overwrites whatever is in `cached-data.csv`
+// because overwrite is set to true.
+
+const csvData = [{x1: 10, x2: 'test1'}, {x1: 20, x2: 'test2'}, ...]
+
+iodide.file.save(csvData, 'cached-data.csv' { overwrite: true })
+
+// iodide.file.save works with ArrayBuffers as well.
+iodide.file.save(new ArrayBuffer(...), 'model-output.bin')
+```
+
+
+
+### `iodide.file.load(fileName, fileType[, variableName])`
+
+Returns a
+[Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises)
+that, when resolved, loads into the notebook the save file `fileName`. For each
+notebook can find a list of uploaded files in the [upload modal](#link-required)
+or on your notebook's [Revisions Page](#needs-link). You can also access any
+uploaded file through the fetch chunk, [following this
+pattern](https://iodide-project.github.io/docs/workflows/#uploading-data-to-an-iodide-notebook).
+For most use cases using [fetch
+chunks](https://iodide-project.github.io/docs/jsmd/#fetch-chunks-fetch) is
+preferable and more straightforward. In more dynamic programmatic cases,
+however, `iodide.file.load` can provide more nuanced workflows.
+
+`fileName` is the name of the file uploaded to the Iodide server. 
+
+`fileType` is the file type to handle. These are the same as the file types in
+[fetch chunks](https://iodide-project.github.io/docs/jsmd/#fetch-chunks-fetch):
+`js` (load this file as javascript, eg a library), `css` (load this file as a
+stylesheet, applying the styles to the report), `json` (load this file as json
```

Comment:
Per your question, I am inclined not to support `js` and `css` for now. I don't see much benefit but lots of potential problems.

Reply:
agreed

Reply:
I'll remove `js` and `css` but we will for sure need file type – otherwise, one would not be able to specify if it is a blob, text, json, or (before long) array buffer.